internal/controllers/gateway/dataplane_mock_test.go

```diff
@@ -0,0 +1,29 @@
+package gateway
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	k8sobj "github.com/kong/kubernetes-ingress-controller/v2/internal/util/kubernetes/object"
+)
+
+type DataplaneMock struct {
+	KubernetesObjectReportsEnabled bool
+	// Mapping namespace to name to status
+	ObjectsStatuses map[string]map[string]k8sobj.ConfigurationStatus
```

Comment:
It's not used in the test added while, was it intended? I think we could use that and verify the `Programmed` condition is set up as well.

Reply:
👍  Can we remove the test cases from the test you're adding then? They seem to differ only by the `KubernetesObjectReportsEnabled` value, but the condition this could affect is not verified in the test anyway. Less code, less indentation, win-win.

Reply:
Fair enough, I've changed it to check for unchanged status when `EnableReferenceGrant` is set to `false.`